doc/source/fq_nmod_sparse_vec.rst

```diff
@@ -0,0 +1,176 @@
+
+    Checks if the given vector is trivial (empty), returning `1` if so and `0` 
+    otherwise
+
+.. function:: void fq_nmod_sparse_vec_equal(const fq_nmod_sparse_vec_t vec1, const fq_nmod_sparse_vec_t vec2, slong ioff, const fq_nmod_ctx_t ctx)
+
+    Checks if vec1 equals vec2 (with s specified column offset ioff), returning
+    `1` if so and `0` otherwise
+
+Indexing
+--------------------------------------------------------------------------------
+
+.. function:: fq_nmod_t * fq_nmod_sparse_vec_at(fq_nmod_sparse_vec_t vec, slong ind, const fq_nmod_ctx_t ctx)
+
+    Returns a pointer to the value at the index ind (or NULL if index not found)
+
+
+Conversion to/from dense vector
+--------------------------------------------------------------------------------
+
+.. function:: void fq_nmod_sparse_vec_from_dense(fq_nmod_sparse_vec_t dst, const fq_nmod_struct *src, slong len, const fq_nmod_ctx_t ctx)
+
+    Converts the dense vector src of length len to a sparse vector
+
+.. function:: void fq_nmod_sparse_vec_to_dense(fq_nmod_struct *dst, const fq_nmod_sparse_vec_t src, slong len, const fq_nmod_ctx_t ctx)
+
+    Converts the sparse vector src to a dense vector of length len
+
+
+Windows, concatenation, and splitting
+--------------------------------------------------------------------------------
+
+.. function:: void fq_nmod_sparse_vec_window_init(fq_nmod_sparse_vec_t window, const fq_nmod_sparse_vec_t vec, slong i1, slong i2, const fq_nmod_ctx_t ctx)
+
+    Constructs a window on a the sparse vector vec between indices i1 and i2
+    Note that window is only valid as long as original vector remains unmodified
+
+.. function:: void fq_nmod_sparse_vec_window_clear(fq_nmod_sparse_vec_t window, const fq_nmod_ctx_t ctx)
+
+    Clears a window (for safety only)
+
+.. function:: void fq_nmod_sparse_vec_concat(fq_nmod_sparse_vec_t res, const fq_nmod_sparse_vec_t vec1, const fq_nmod_sparse_vec_t vec2, slong len1, const fq_nmod_ctx_t ctx)
+
+    Concatenates two vectors vec1 and vec2 into res, with indices of vec2 
+    offset by len1
+
+.. function:: void fq_nmod_sparse_vec_split(fq_nmod_sparse_vec_t res1, fq_nmod_sparse_vec_t res1, const fq_nmod_sparse_vec_t vec, slong ind, const fq_nmod_ctx_t ctx)
+
+    Splits vec into two vectors res1 and res2, with res1 containing all entries 
+    below index ind and res2 containing the rest
+
+Permutation
+--------------------------------------------------------------------------------
+
+.. function:: void fq_nmod_sparse_vec_permute_inds(fq_nmod_sparse_vec_t vec, slong *P, const fq_nmod_ctx_t ctx)
+
+    Permutes the indices of vec according to P, and resorts
+
+
+Randomization
+--------------------------------------------------------------------------------
+
+
+.. function:: void fq_nmod_sparse_vec_randtest(fq_nmod_sparse_vec_t vec, flint_rand_t state, slong nnz, slong len, const fq_nmod_ctx_t ctx)
+
+    Makes vec a sparse vector with nnz nonzero entries uniformly distributed
```

Comment:
Best not to use the words "uniformly distributed" as this is very hard to guarantee. "randomly distributed" is more vague.